phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -867,7 +868,9 @@ private void addColumnToTable(List<Cell> results, PName colName, PName famName,
         PDataType dataType =
                 PDataType.fromTypeId(PInteger.INSTANCE.getCodec().decodeInt(
                         dataTypeKv.getValueArray(), dataTypeKv.getValueOffset(), SortOrder.getDefault()));
-        if (maxLength == null && dataType == PBinary.INSTANCE) dataType = PVarbinary.INSTANCE;   // For
+        if (maxLength == null && dataType == PBinary.INSTANCE) {
```

Comment:
nit: formatting of comment is mixed up now. (should be "for backward compatibility")